### DISCUSSION TITLE
Get an Image ByteArray

### DISCUSSION THREAD

**Discussion by HEIC-to-JPEG-Dev** [dlemstra/Magick.NET#1409] [Help] [answered]
What would be a good way to get an image as a byte array. I looked at ToByteArray, but that includes all the profile/exif stuff in addition to the image.

Would removing the exif profile, then ToByteArray work or would it still contain more than just the image ?
Or should I be looking at GetPixels and somehow convert that ?

End goal here is to get a unique sha256 hash of the image.

Thanks

**Comment by dlemstra** [comment, marked as answer]
I think you could use `image.Signature` for this? This only looks at the pixels off the image.

**Reply by HEIC-to-JPEG-Dev** [reply]
Thank you. I can use that. Is there a way to get the signature as a uLong ?

**Reply by dlemstra** [reply]
The 256 bits of this signature will not fit inside 64 bits so you cannot change this to an `ulong`.

**Reply by HEIC-to-JPEG-Dev** [reply]
That’s fine, I hash the signature to produce a uLong :)
